kafka-1.0.x/src/main/scala/monix/kafka/Commit.scala

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) 2014-2019 by The Monix Project Developers.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package monix.kafka
+
+import monix.eval.Task
+import org.apache.kafka.common.TopicPartition
+import org.apache.kafka.clients.consumer.OffsetCommitCallback
+
+/**
+  * Callback for batched commit realized as closure
+  * in [[KafkaConsumerObservable]] context. This decision was made for
+  * thread-safety reasons.
+  * */
+trait Commit {
+  def commitBatchSync(batch: Map[TopicPartition, Long]): Task[Unit]
+  def commitBatchAsync(batch: Map[TopicPartition, Long]): Task[Unit]
+  def commitBatchAsync(batch: Map[TopicPartition, Long], callback: OffsetCommitCallback): Task[Unit]
```

Comment:
I think we could have just two methods with default parameter, right? Either `null` or `Option`

Reply:
I tried to make it similar with original kafka client api:

```java
    public void commitAsync() {
        commitAsync(null);
    }
    public void commitAsync(OffsetCommitCallback callback)
```

And in my mind overloaded methods are more readable than one method with default `null`. But I forgot to make one of this methods final:

```scala
trait Commit {
  def commitBatchSync(batch: Map[TopicPartition, Long]): Task[Unit]
  def commitBatchAsync(batch: Map[TopicPartition, Long], callback: OffsetCommitCallback): Task[Unit]
  final def commitBatchAsync(batch: Map[TopicPartition, Long]): Task[Unit] = commitBatchAsync(batch, null)
}
```

What do you think about this version?

Reply:
Looks good

Reply:
Done